### PR TITLE
CI Update Python

### PR DESCRIPTION
**Pull request by ahartikainen**
## Description

Update Python versions. Drop 3.6 checks.

New default Python is 3.9.

### CODE REVIEW

**Comment by OriolAbril** [comment]
We should also update the readme: https://github.com/arviz-devs/arviz#dependencies and the setup.py: https://github.com/arviz-devs/arviz/blob/master/setup.py#L69-L71.

File: scripts/create_testenv.sh [hunk 1]
```diff
@@ -7,8 +7,8 @@ command -v conda >/dev/null 2>&1 || {
   exit 1;
 }
 
-# if no python specified, use Travis version, or else 3.8
-PYTHON_VERSION=${PYTHON_VERSION:-${TRAVIS_PYTHON_VERSION:-3.8}}
+# if no python specified, use Travis version, or else 3.9
+PYTHON_VERSION=${PYTHON_VERSION:-${TRAVIS_PYTHON_VERSION:-3.9}}
```
Comment (canyon289) [inline scripts/create_testenv.sh:11]: We should probably rename from TRAVIS_PYTHON_VERSION to something else since were not using TravisCI anymore

**Comment by canyon289** [comment]
Rebased on master

**Comment by canyon289** [comment]
Tensorflow is not available on 3.9 yet https://github.com/tensorflow/tensorflow/issues/44485

**Comment by ahartikainen** [comment]
Neither is pystan3

**Comment by canyon289** [comment]
Got it, should we keep PR on hold until then?

**Comment by ahartikainen** [comment]
Yes (and clean it a bit)

**Comment by OriolAbril** [comment]
Any news on that? Should we merge and drop python 3.6 in the coming release or wait for the next one? @canyon289 @ahartikainen

**Comment by ahartikainen** [comment]
Wait. Some packages are still missing.

**Comment by OriolAbril** [comment]
I think we should drop python 3.6 in the next release (most pydata packages have already done). I am adding it to the milestone, feel free to remove if you think we still should wait

**Comment by OriolAbril** [comment]
Just for future reference, numpy has already stopped 3.7 support, so not too long from now once numba supports 3.10 we should change 3.7 for 3.10. Should be ready to merge now

**Comment by OriolAbril** [comment]
seems like I can't tag @ahartikainen for review as it's his PR

**Comment by ahartikainen** [comment]
Lgtm

**Pull request merged**